src/rust/zh.manhuagui/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+#![no_std]
+extern crate alloc;
+
+mod decoder;
+mod helper;
+mod parser;
+
+use aidoku::{
+	error::Result, prelude::*, std::net::HttpMethod, std::net::Request, std::String, std::Vec,
+	Chapter, Filter, Manga, MangaPageResult, Page,
+};
+
+const BASE_URL: &str = "https://www.manhuagui.com";
+// const MOBILE_URL: &str = "https://m.manhuagui.com";
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut url = String::new();
+
+	parser::get_filtered_url(filters, page, &mut url);
+
+	let html = Request::new(url.as_str(), HttpMethod::Get).html()?;
+	if url.contains("https://www.manhuagui.com/list/") {
+		let result = parser::parse_home_page(html).unwrap();
+		return Ok(result);
```

Comment:
`parse_home_page` returns a result so you can just return that
```suggestion
		return parser::parse_home_page(html);
```